controllers/temporalappworker_controller.go

```diff
@@ -0,0 +1,215 @@
+	if updated {
+		err := r.Update(ctx, appworker)
+		if err != nil {
+			logger.Error(err, "Can't set cluster defaults")
+			return r.handleError(ctx, appworker, "", err)
+		}
+		// As we updated the instance, another reconcile will be triggered.
+		return reconcile.Result{}, nil
+	}
+
+	if err := r.reconcileResources(ctx, appworker); err != nil {
+		logger.Error(err, "Can't reconcile resources")
+		return r.handleErrorWithRequeue(ctx, appworker, v1beta1.ResourcesReconciliationFailedReason, err, 2*time.Second)
+	}
+
+	return r.handleSuccess(ctx, appworker)
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *TemporalAppWorkerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&temporaliov1beta1.TemporalAppWorker{}).
+		Complete(r)
+}
+
+func (r *TemporalAppWorkerReconciler) handleErrorWithRequeue(ctx context.Context, appworker *v1beta1.TemporalAppWorker, reason string, err error, requeueAfter time.Duration) (ctrl.Result, error) {
+	if reason == "" {
+		reason = v1beta1.ReconcileErrorReason
+	}
+	v1beta1.SetTemporalAppWorkerReconcileError(appworker, metav1.ConditionTrue, reason, err.Error())
+	err = r.updateClusterStatus(ctx, appworker)
+	return reconcile.Result{RequeueAfter: requeueAfter}, err
+}
+
+func (r *TemporalAppWorkerReconciler) handleError(ctx context.Context, appworker *v1beta1.TemporalAppWorker, reason string, err error) (ctrl.Result, error) {
+	return r.handleErrorWithRequeue(ctx, appworker, reason, err, 0)
+}
+
+func (r *TemporalAppWorkerReconciler) updateClusterStatus(ctx context.Context, cluster *v1beta1.TemporalAppWorker) error {
+	err := r.Status().Update(ctx, cluster)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *TemporalAppWorkerReconciler) reconcileResources(ctx context.Context, temporalAppWorker *v1beta1.TemporalAppWorker) error {
+	logger := log.FromContext(ctx)
+
+	appWorkerBuilder := appworker.ClusterBuilder{
+		Instance: temporalAppWorker,
+		Scheme:   r.Scheme,
+	}
+
+	builders, err := appWorkerBuilder.ResourceBuilders()
+	if err != nil {
+		return err
+	}
+
+	logger.Info("Retrieved builders", "count", len(builders))
+
+	for _, builder := range builders {
+		if comparer, ok := builder.(resource.Comparer); ok {
+			err := equality.Semantic.AddFunc(comparer.Equal)
+			if err != nil {
+				return err
+			}
+		}
+
+		res, err := builder.Build()
+		if err != nil {
+			return err
+		}
+
+		operationResult, err := controllerutil.CreateOrUpdate(ctx, r.Client, res, func() error {
+			return builder.Update(res)
+		})
+		r.logAndRecordOperationResult(ctx, temporalAppWorker, res, operationResult, err)
+		if err != nil {
+			return err
+		}
+	}
+
+	return r.updateClusterStatus(ctx, temporalAppWorker)
+}
+
+func (r *TemporalAppWorkerReconciler) logAndRecordOperationResult(ctx context.Context, appworker *v1beta1.TemporalAppWorker, resource runtime.Object, operationResult controllerutil.OperationResult, err error) {
```

Comment:
I think that we can factorize this function with the same one in the cluster reconcilier. 
Maybe this function signature can work:
```suggestion
func (r *TemporalAppWorkerReconciler) logAndRecordOperationResult(ctx context.Context, appworker *runtime.Object, resource runtime.Object, operationResult controllerutil.OperationResult, err error) {
```

Reply:
@alexandrevilain it doesn't work its a casting issue